py/server/deephaven/parquet.py

```diff
@@ -419,8 +397,7 @@ def write_partitioned(
 def batch_write(
     tables: List[Table],
     paths: List[str],
-    table_definition: Optional[Union[Dict[str, DType], List[Column]]] = None,
-    col_definitions: Optional[List[Column]] = None,
+    table_definition: Union[Dict[str, DType], List[Column]],
```

Comment:
why not None?

Reply:
So after this change, table_definition will become a mandatory parameter.
Earlier, user had to supply either `table_definition` or `col_definitions`. There was a check verifying this in the implementation of `batch_write`. Now we have removed `col_definitions`, so `table_definition` has become mandatory.

Reply:
You are right, we do pull the definition for regular `read` and `write` calls. The problem is with `batch_write` where we are writing more than one tables. So then user needs to provide the definition because we cannot decide for them.
One approach can be to verify if all tables have same definition and then use that definition by default. But the java code also currently expects a definition for `batch_write`. So did not want to change the API in this PR.

Reply:
Verifying they are all the same is the right thing to do.  That can happen in python without changing the java.

Reply:
I meant that I wanted to keep the APIs consistent in both python and Java. That's what was happening earlier as well and this is not something new in this PR.
Earlier, users had to provide either `table_definition` or `col_definitions` when using `batch_write`. Now we have removed `col_definitions`, so `table_definition` has become mandatory.

I can discuss with Ryan about making this field optional in case all tables have same definition or if user provides a single table, and do that as a separate PR.

Reply:
I'm ok with it going in as is, but the UX is not good and needs a follow-on ticket to address it.